usr/lib/common/utility.c

```diff
@@ -581,9 +581,10 @@ CK_RV compute_next_token_obj_name(CK_BYTE *current, CK_BYTE *next)
     for (i = 0; i < 8; i++) {
         if (current[i] >= '0' && current[i] <= '9')
             val[i] = current[i] - '0';
-
-        if (current[i] >= 'A' && current[i] <= 'Z')
+        else if (current[i] >= 'A' && current[i] <= 'Z')
             val[i] = current[i] - 'A' + 10;
+        else
+            val[i] = 36;
```

Comment:
The function should fail here with e.g. EINVAL because a byte in current is not in range [0-9A-Z].

Reply:
Yes I agree, but not all callers check the return code of this function (e.g.  object_mgr_copy() and object_mgr_create_final() so I thought it would be better to do something that produces a valid name. With value 36 the code following this will set the invalid character to '0' and increment the following characters by one to make in unique. 

Let me know if you still prefer to return an error here, with the risk that the calling function will ignore the return code and work with a non-unique or invalid object name.

Reply:
Okay so leave it like that but extend the above comment stating that function will return the next obj name even if given name is invalid.

Reply:
done.